homeassistant/components/switchbot_cloud/__init__.py

```diff
@@ -136,6 +138,18 @@ async def make_device_data(
         )
         devices_data.locks.append((device, coordinator))
 
+    if isinstance(device, Device) and device.device_type in ["Bot"]:
+        coordinator = await coordinator_for_device(
+            hass, api, device, coordinators_by_id
+        )
+        if (
+            coordinator.data is not None
+            and coordinator.data.get("deviceMode") == "pressMode"
+        ):
+            devices_data.buttons.append((device, coordinator))
+        else:
+            devices_data.switches.append((device, coordinator))
```

Comment:
if `coordinator.data is None` we add it to `switches`. should we move that check to another `if` instead ?

Reply:
I've moved the check to its own `if`.